### PR TITLE
vecgeom: add v1.2.9

### PR DESCRIPTION
**Pull request by sethrj**

### CODE REVIEW

File: var/spack/repos/builtin/packages/vecgeom/package.py [hunk 1]
```diff
@@ -21,6 +21,11 @@ class Vecgeom(CMakePackage, CudaPackage):
     maintainers("drbenmorgan", "sethrj")
 
     version("master", branch="master")
+    version(
+        "1.2.9",
+        url="https://gitlab.cern.ch/-/project/981/uploads/55a89cafbf48a418bec68be42867d4bf/VecGeom-v1.2.9.tar.gz",
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/vecgeom/package.py:26]: Different url than other versions. Does the previous schema still work?

**Review by bendamasio — COMMENTED** [review]
Hello @drbenmorgan, I generated some build results:<br>
:information_source: `spack install` on the changed recipes of this PR was successful[^1] on Ubuntu 22.04.4 LTS!<details><summary>Click here for a summary of the successful builds.</summary>
These changed specs were found using [gh pr diff](https://cli.github.com):
- `vecgeom@1.2.9`


<details><summary><b>Click here to show/hide the report on </b><tt>spack find</tt> for <tt>linux-ubuntu22.04-skylake / gcc@13.1.0</tt>.</summary><ol>

<li><details><summary>Click on this area to unfold/refold the dependency report of this build:

>```py
>vecgeom@1.2.9~cuda+gdml~geant4~ipo~root+shared build_system=cmake build_type=Release cxxstd=17 generator=make
>```
</summary>

>```py
>	cmake@3.30.5
>		curl@8.10.1 tls=openssl
>			nghttp2@1.63.0
>				diffutils@3.10
>			openssl@3.3.1 certs=mozilla
>				ca-certificates-mozilla@2023-05-30
>				perl@5.40.0cpanmopcodeopenthreads
>					berkeley-db@18.1.40cxx,b231fcc
>					bzip2@1.0.8
>					gdbm@1.23
>						readline@8.2
>			pkgconf@2.2.0
>		ncurses@6.5
>		zlib-ng@2.2.1compatnew_strategiesopt
>	gcc-runtime@13.1.0
>	glibc@2.35
>	gmake@4.4.1
>	veccore@0.8.1
>	xerces-c@3.2.5 cxxstd=17 netaccessor=curl transcoder=gnuiconv
>		libiconv@1.17
>			gmake@4.4.1
>
>==> 1 installed package
>```

</details>



</details>

[^1]: Orchestrated using `spack install` by [github.com/spack/gh-spack-pr](https://github.com/spack/gh-spack-pr) and the [GitHub CLI](https://cli.github.com/)
</details><details><summary>Click here for a disclaimer for the maintainers of the recipes

</summary><br>👍 Hello @drbenmorgan!<br><br>

- We would like to ask you if you could review this PR within 5 days.
- If the PR is not reviewed within 5 days, the PR can be approved and merged by members.
- Please use 👍, Approve, LGTM or Request the comment so we can act accordingly.
- If you like to review but need more time, please add a comment to the PR.
- If you have questions (or would like to be removed as maintainer), please let us know.

Thank you for your time and effort!<br>
You can also join the Spack Slack channel for more information and have a chat:
https://spackpm.slack.com → #pull-requests. Thank you and best regards, @x64</details>

**Review by bendamasio — COMMENTED** [review]
Hello @drbenmorgan, I generated some build results:<br>
:information_source: `spack install` on the changed recipes of this PR was successful[^1] on Ubuntu 22.04.4 LTS!<details><summary>Click here for a summary of the successful builds.</summary>
These changed specs were found using [gh pr diff](https://cli.github.com):
- `vecgeom@1.2.9`
- `vecgeom@1.2.9`
- `vecgeom@1.2.8`
- `vecgeom@1.2.7`


<details><summary><b>Click here to show/hide the report on </b><tt>spack find</tt> for <tt>linux-ubuntu22.04-skylake / gcc@13.1.0</tt>.</summary><ol>

<li><details><summary>Click on this area to unfold/refold the dependency report of this build:

>```py
>vecgeom@1.2.7~cuda+gdml~geant4~ipo~root+shared build_system=cmake build_type=Release cxxstd=17 generator=make
>```
</summary>

>```py
>	cmake@3.30.5
>		curl@8.10.1 tls=openssl
>			nghttp2@1.63.0
>				diffutils@3.10
>			openssl@3.3.1 certs=mozilla
>				ca-certificates-mozilla@2023-05-30
>				perl@5.40.0cpanmopcodeopenthreads
>					berkeley-db@18.1.40cxx,b231fcc
>					bzip2@1.0.8
>					gdbm@1.23
>						readline@8.2
>			pkgconf@2.2.0
>		ncurses@6.5
>		zlib-ng@2.2.1compatnew_strategiesopt
>	gcc-runtime@13.1.0
>	glibc@2.35
>	gmake@4.4.1
>	veccore@0.8.1
>	xerces-c@3.2.5 cxxstd=17 netaccessor=curl transcoder=gnuiconv
>		libiconv@1.17
>			gmake@4.4.1
>
>```

</details>


<li><details><summary>Click on this area to unfold/refold the dependency report of this build:

>```py
>vecgeom@1.2.8~cuda+gdml~geant4~ipo~root+shared build_system=cmake build_type=Release cxxstd=17 generator=make
>```
</summary>

>```py
>	cmake@3.30.5
>		curl@8.10.1 tls=openssl
>			nghttp2@1.63.0
>				diffutils@3.10
>			openssl@3.3.1 certs=mozilla
>				ca-certificates-mozilla@2023-05-30
>				perl@5.40.0cpanmopcodeopenthreads
>					berkeley-db@18.1.40cxx,b231fcc
>					bzip2@1.0.8
>					gdbm@1.23
>						readline@8.2
>			pkgconf@2.2.0
>		ncurses@6.5
>		zlib-ng@2.2.1compatnew_strategiesopt
>	gcc-runtime@13.1.0
>	glibc@2.35
>	gmake@4.4.1
>	veccore@0.8.1
>	xerces-c@3.2.5 cxxstd=17 netaccessor=curl transcoder=gnuiconv
>		libiconv@1.17
>			gmake@4.4.1
>
>```

</details>


<li><details><summary>Click on this area to unfold/refold the dependency report of this build:

>```py
>vecgeom@1.2.9~cuda+gdml~geant4~ipo~root+shared build_system=cmake build_type=Release cxxstd=17 generator=make
>```
</summary>

>```py
>	cmake@3.30.5
>		curl@8.10.1 tls=openssl
>			nghttp2@1.63.0
>				diffutils@3.10
>			openssl@3.3.1 certs=mozilla
>				ca-certificates-mozilla@2023-05-30
>				perl@5.40.0cpanmopcodeopenthreads
>					berkeley-db@18.1.40cxx,b231fcc
>					bzip2@1.0.8
>					gdbm@1.23
>						readline@8.2
>			pkgconf@2.2.0
>		ncurses@6.5
>		zlib-ng@2.2.1compatnew_strategiesopt
>	gcc-runtime@13.1.0
>	glibc@2.35
>	gmake@4.4.1
>	veccore@0.8.1
>	xerces-c@3.2.5 cxxstd=17 netaccessor=curl transcoder=gnuiconv
>		libiconv@1.17
>			gmake@4.4.1
>
>```

</details>



</details>

[^1]: Orchestrated using `spack install` by [github.com/spack/gh-spack-pr](https://github.com/spack/gh-spack-pr) and the [GitHub CLI](https://cli.github.com/)
</details><details><summary>Click here for a disclaimer for the maintainers of the recipes

</summary><br>👍 Hello @drbenmorgan!<br><br>

- We would like to ask you if you could review this PR within 5 days.
- If the PR is not reviewed within 5 days, the PR can be approved and merged by members.
- Please use 👍, Approve, LGTM or Request the comment so we can act accordingly.
- If you like to review but need more time, please add a comment to the PR.
- If you have questions (or would like to be removed as maintainer), please let us know.

Thank you for your time and effort!<br>
You can also join the Spack Slack channel for more information and have a chat:
https://spackpm.slack.com → #pull-requests. Thank you and best regards, @x64</details>

File: var/spack/repos/builtin/packages/vecgeom/package.py (hunk 1 above)
Comment (drbenmorgan) [reply var/spack/repos/builtin/packages/vecgeom/package.py:26]: I've checked and the `https://gitlab.cern.ch/VecGeom/VecGeom/uploads` urls still give a download. I _think_ it's just a change to how GitLab renders these, and the old ones remain active, or forward.

**Review by drbenmorgan — APPROVED** [review]
Looks good to me, so LGTM

**Pull request merged**